method/link.js

```diff
@@ -84,6 +84,50 @@ module.exports = function Link (ssb) {
         }),
         pull.collect(cb)
       )
+    },
+    findSubgroupByGroupId (groupId, cb) {
```

Comment:
@mixmix do you think i should reduce this code abit? If yes, any suggestions? This method and the one above it are really similar

Reply:
I think it's fine.

If you REALLY want to extract something you can do

```js
function getLinks (query, isRoot, cb) {
      pull(
        ssb.backlinks.read({ query }),
        pull.unique(link => link.value.content.child),
        pull.filter(isRoot),
        pull.filter(link => {
          return link.value.content.parent === link.value.content.recps[0]
          // it would be very strange for a link to be created like this
          // but we should consider it unsafe and ignore it I think
        }),
        pull.map(link => {
          const { child, recps } = link.value.content

          return {
            linkId: link.key,
            groupId,
            subgroupId: child,
            recps,
            states: [{
              head: link.key,
              state: {
              }
            }]
          }
        }),
        pull.collect(cb)
      )
}
```

Reply:
OHHH, I remember why this was `ssb.query.read` (using ssb-query).
the backlinks query which get EVERY message which contains the `groupId` in it.. which is gonna be a LOT for a group, then narrows those down.

the ssb.query on the other hand will use and index on `type` and there will presumably NOT be that many link/group-subgroup messages... at least less of those messages than are in the parent group